bundles/org.openhab.binding.gce/src/main/java/org/openhab/binding/gce/internal/handler/Ipx800DeviceConnector.java

```diff
@@ -72,12 +75,15 @@ public synchronized void send(String message) {
      */
     private void connect() throws IOException {
         disconnect();
+
         logger.debug("Connecting {}:{}...", hostname, portNumber);
-        client = new Socket(hostname, portNumber);
+        Socket client = new Socket(hostname, portNumber);
         client.setSoTimeout(DEFAULT_SOCKET_TIMEOUT_MS);
         client.getInputStream().skip(client.getInputStream().available());
-        in = new BufferedReader(new InputStreamReader(client.getInputStream()));
-        out = new PrintWriter(client.getOutputStream(), true);
+        socket = Optional.of(client);
```

Comment:
```suggestion
        this.socket = Optional.of(socket);
```

Comes together with the previous comment

Reply:
Corrected accordingly.